### ISSUE TITLE
Question: KSP Processor Priority

### ISSUE THREAD

**Issue by tamimattafi** [google/ksp#1043]
Hello! Thank you first this awesome API!
However, I'm trying to generate some code that is going to be used by other processors to generate other code. Here's a link to the processor: https://github.com/tamimattafi/android-room-compound

Everything works fine if you clean and rebuild the project, but sometimes when you try to run the app, other annotation processors try to generate their code, but mine isn't generated yet, and it gives me an unresolved symbol exception. Is there a way to give my processor more priority? So it always runs first, before any other processors?

**Comment by neetopia** [comment]
It is not possible to specify a priory for processors in terms of execution ordering. Instead, you should return the symbols that you are not able to process to next round in your processor's return result. Please take a look at our multiple round processing document for more detail.

**Comment by tamimattafi** [comment]
@neetopia Thank you for your reply!
However, the problem is, the processor that's failing isn't my own processor, so I'm not able to implement round processing, I'm generating code, that is going to be used by the other processor to generate other code, and sometimes, my code isn't yet generated, and the other generator fails because it can't see that code.

Sad that there's no way to regulate priority or queue processors in a specific order:(

**Comment by neetopia** [comment]
In that case, it seems like a bug on the other processor, maybe better to contact the author to get this behavior fixed.

**Comment by NorbertSandor** [comment]
My use case:
1. `ProcessorA` generates class `a.b.ClassA`
2. `ProcessorB` enumerates declarations in package `a.b` using `com.google.devtools.ksp.processing.Resolver#getDeclarationsFromPackage` and generates one aggregated class `a.b.Aggregated` from them.

If `ProcessorB` is executed before `ProcessorA` then `ProcessorB` works with incomplete input. \
I tried to cache data about the classes processed in round#1 in `ProcessorB` but when I try to regenerate `a.b.Aggregated` in round#2 then a `kotlin.io.FileAlreadyExistsException: .../Aggregated.kt` exception is thrown.

**Comment by danysantiago** [comment]
Multi-round must be cooperative so other 3rd party processors need to also implement deferring. Note that KAPT (javac really) uses exactly the same technique as KSP, it will continue processing until no new sources are generated and processors that find unresolved types tend to defer their processing in hopes for another processor to generate them.

I'm curious about your Dagger use-case because they do support multi-round processing. If you defined a `MyGenerateModule` in source and use it in a Dagger `@Component`, Dagger will not generate the component code in the first round because it knows `MyGenerateModule` is an unresolved type, it will hope that your processor will generate it in round 1 and then in round 2 Dagger will be able to consume it and generate the component code.

In my opinion https://github.com/google/ksp/issues/1981 looks like a user-error, while https://github.com/google/ksp/issues/1912 is unrelated, SQLDelight generates code via a Gradle Plugin not a KSP processor, so the wiring of those is very different and are not part of the multi-round / multi-processor collaboration subject this issue is about.

**Comment by danysantiago** [comment]
Is your `@MyAnnotation` also annotated with Hilt's `@GeneratesRootInput`? This is important for Hilt's aggregation mechanism to work and discover generated `@InstallIn` modules, see https://dagger.dev/hilt/creating-extensions.html

**Comment by rolgalan** [comment]
No, I am not annotating that annotation with `@GeneratesRootInput` and still it works fine with Dagger over kapt.

I just tried adding it to `@MyAnnotation` but nothing is generated by Dagger KSP neither.

Checking the documentation provided I also see a mention to `@OriginatingElement`. I was not using this neither. I also tried adding this into my generated module `@OriginatingElement(topLevelClass = MyAnnotation::class)` and still nothing works (but not sure if that should be the topLevelClass).
 
We will review how auto-dagger operates to see if we have something wrong in our code, as based on your responses this should work fine out of the box 🤷.

Thanks for your support in any case!
